apistar/server/asgi.py

```diff
@@ -94,23 +95,28 @@ def resolve(self,
 class HeaderComponent(Component):
     def resolve(self,
                 parameter: Parameter,
-                headers: http.Headers) -> http.Header:
+                headers: http.Headers) -> typing.Optional[http.Header]:
```

Comment:
These won’t work as expected, since ‘Component’ takes its default from the return annotation. We might want to change that in favour of more explicit behaviour, but either way we can simply make this change, as it’ll stop ‘Header’ being provided.